internal/controller/api_to_config_test.go

```diff
@@ -1586,10 +1601,314 @@ func TestConversion(t *testing.T) {
+											BFDProfile: "bfd2",
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				{
+					Spec: v1beta1.FRRConfigurationSpec{
+						BGP: v1beta1.BGPConfig{
+							BFDProfiles: []v1beta1.BFDProfile{
+								{
+									Name:             "bfd1",
+									ReceiveInterval:  42,
+									TransmitInterval: 43,
+								},
+							},
+						},
+					},
+				},
+			},
+			secrets:  map[string]v1.Secret{},
+			expected: nil,
+			err:      errors.New("got failed to process neighbor 65041@192.0.2.21 for router 65040-: neighbor 65041@192.0.2.21 referencing non existing BFDProfile bfd2"),
+		},
+		{
+			name: "Two BFDProfiles, but identical",
+			fromK8s: []v1beta1.FRRConfiguration{
+				{
+					Spec: v1beta1.FRRConfigurationSpec{
+						BGP: v1beta1.BGPConfig{
+							BFDProfiles: []v1beta1.BFDProfile{
+								{
+									Name:             "bfd1",
+									ReceiveInterval:  42,
+									TransmitInterval: 43,
+								},
+								{
+									Name:             "bfd1",
+									ReceiveInterval:  42,
+									TransmitInterval: 43,
+								},
+								{
+									Name:             "bfd3",
+									ReceiveInterval:  42,
+									TransmitInterval: 44,
+								},
+							},
+						},
+					},
+				},
+			},
+			secrets:  map[string]v1.Secret{},
+			expected: nil,
+			err:      errors.New("duplicate bfd profile name bfd1"),
+		},
+		{
+			name: "Two BFDProfiles, from different configs",
+			fromK8s: []v1beta1.FRRConfiguration{
+				{
+					Spec: v1beta1.FRRConfigurationSpec{
+						BGP: v1beta1.BGPConfig{
+							BFDProfiles: []v1beta1.BFDProfile{
+								{
+									Name:             "bfd1",
+									ReceiveInterval:  42,
+									TransmitInterval: 43,
+								},
+								{
+									Name:             "bfd2",
+									ReceiveInterval:  42,
+									TransmitInterval: 44,
+								},
+							},
+						},
+					},
+				},
+				{
+					Spec: v1beta1.FRRConfigurationSpec{
+						BGP: v1beta1.BGPConfig{
+							BFDProfiles: []v1beta1.BFDProfile{
+								{
+									Name:             "bfd1",
+									ReceiveInterval:  42,
+									TransmitInterval: 43,
+								},
+								{
+									Name:             "bfd3",
+									ReceiveInterval:  42,
+									TransmitInterval: 45,
+								},
+							},
+						},
+					},
+				},
+			},
+			secrets: map[string]v1.Secret{},
+			expected: &frr.Config{
+				Routers: []*frr.RouterConfig{},
+				BFDProfiles: []frr.BFDProfile{
+					{
+						Name:             "bfd1",
+						ReceiveInterval:  pointer.Uint32Ptr(42),
+						TransmitInterval: pointer.Uint32Ptr(43),
+					},
+					{
+						Name:             "bfd2",
+						ReceiveInterval:  pointer.Uint32Ptr(42),
+						TransmitInterval: pointer.Uint32Ptr(44),
+					},
+					{
+						Name:             "bfd3",
+						ReceiveInterval:  pointer.Uint32Ptr(42),
+						TransmitInterval: pointer.Uint32Ptr(45),
+					},
+				},
+			},
+			err: nil,
+		},
+		{
+			name: "Two BFDProfiles, from different configs and different values",
```

Comment:
I think this is redundant with the previous one

Reply:
removed